include/knowhere/index/handle.h

```diff
@@ -0,0 +1,60 @@
+// Copyright (C) 2019-2023 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License.
+#ifndef KNOWHERE_HANDLE_H
+#define KNOWHERE_HANDLE_H
+#include <atomic>
+#include <chrono>
+#include <memory>
+
+#include "knowhere/expected.h"
+
+namespace folly {
+template <typename T>
+class Future;
+
+}
+
+namespace knowhere {
+class Handle {
```

Comment:
would it be possible to give a more specific name? `Handle` is a very generic one. Maybe, make the class name contain a word `Interrupt` or `Timeout`?

Reply:
The changes have been made.